src/components/CallView/shared/ReactionToaster.vue

```diff
@@ -0,0 +1,165 @@
+<!--
+  - @copyright Copyright (c) 2023 Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @author Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<transition-group name="toast" class="toaster" tag="ul">
+		<li v-for="toast in toasts"
+			:key="toast.seed"
+			class="toast"
+			:style="styled(toast.user, toast.seed)">
+			<span class="toast__reaction">
+				{{ toast.reaction }}
+			</span>
+			<span class="toast__name">
+				{{ toast.user }}
+			</span>
+		</li>
+	</transition-group>
+</template>
+
+<script>
+import usernameToColor from '@nextcloud/vue/dist/Functions/usernameToColor.js'
+
+import { EventBus } from '../../../services/EventBus.js'
+import { throttleQueue } from '../../../utils/throttle.js'
+
+export default {
+	name: 'ReactionToaster',
+
+	props: {
+		/**
+		 * The conversation token
+		 */
+		token: {
+			type: String,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			animationLength: 2000,
+			toasts: [],
+		}
+	},
+
+	mounted() {
+		EventBus.$on('send-reaction', throttleQueue(this.showReaction, this.animationLength / 4))
+	},
+
+	beforeDestroy() {
+		EventBus.$off('send-reaction')
+	},
+
+	methods: {
+		showReaction({ user, reaction }) {
+			const seed = Math.random()
+
+			this.toasts.push({ seed, reaction, user })
+
+			setTimeout(() => {
+				this.toasts.shift()
+			}, this.animationLength)
+		},
+
+		styled(user, seed) {
+			const styles = {}
+
+			const color = usernameToColor(user)
+			styles['--background-color'] = `rgb(${color.r}, ${color.g}, ${color.b})`
+			styles['--animation-length'] = `${this.animationLength + 300}ms`
+			styles['--horizontal-offset'] = `${5 + 10 * seed}%`
```

Comment:
not sure which one is which, but I think the left-right should vary more. 5 pixels or 10% is too little. I would at least allow for a left-right offset of 30-40%
```suggestion
			styles['--horizontal-offset'] = `${5 + 30 * seed}%`
```